src/index.js

```diff
@@ -93,69 +74,139 @@ export async function createWindowsInstaller(options) {
-  if (path.sep === '/') {
-    templateData = templateData.replace(/\\/g, '/');
-  }
-  const nuspecContent = template(templateData)(metadata);
-
-  log(`Created NuSpec file:\n${nuspecContent}`);
+  const outputDirectory = path.resolve(options.outputDirectory || 'installer');
+  if (options.remoteReleases) {
+    let cmd = path.join(vendorPath, 'SyncReleases.exe');
+    let args = ['-u', options.remoteReleases, '-r', outputDirectory];
 
-  const nugetOutput = await fsUtils.createTempDir('si-');
-  const targetNuspecPath = path.join(nugetOutput, metadata.name + '.nuspec');
-
-  await fsUtils.writeFile(targetNuspecPath, nuspecContent);
+    if (useMono) {
+      args.unshift(cmd);
+      cmd = 'mono';
+    }
 
-  let cmd = path.join(vendorPath, 'nuget.exe');
-  let args = [
-    'pack', targetNuspecPath,
-    '-BasePath', appDirectory,
-    '-OutputDirectory', nugetOutput,
-    '-NoDefaultExcludes'
-  ];
+    if (options.remoteToken) {
+      args.push('-t', options.remoteToken);
+    }
 
-  if (useMono) {
-    args.unshift(cmd);
-    cmd = monoExe;
+    await spawn(cmd, args);
   }
 
-  // Call NuGet to create our package
-  log(await spawn(cmd, args));
-  const nupkgPath = path.join(nugetOutput, `${metadata.name}.${metadata.version}.nupkg`);
+  // todo fix Squirrel.windows "Sharing violation on path" (avoid copy, use file directly)
+  const squirrelWorkaroundDir = path.join(outputDirectory, '.tmp');
+  await fsUtils.mkdirs(squirrelWorkaroundDir);
+  try {
+    const nupkgPath = path.join(squirrelWorkaroundDir, 'in.nupkg');
+    await pack(metadata, appDirectory, nupkgPath);
+    await releasify(nupkgPath, outputDirectory, options, vendorPath);
+  }
+  finally {
+    await fsUtils.remove(squirrelWorkaroundDir);
+  }
 
-  if (remoteReleases) {
-    cmd = path.join(vendorPath, 'SyncReleases.exe');
-    args = ['-u', remoteReleases, '-r', outputDirectory];
+  if (options.fixUpPaths !== false) {
+    log('Fixing up paths');
 
-    if (useMono) {
-      args.unshift(cmd);
-      cmd = monoExe;
+    if (metadata.productName || options.setupExe) {
+      const setupPath = path.join(outputDirectory, options.setupExe || `${metadata.productName}Setup.exe`);
+      const unfixedSetupPath = path.join(outputDirectory, 'Setup.exe');
+      log(`Renaming ${unfixedSetupPath} => ${setupPath}`);
+      await fsUtils.rename(unfixedSetupPath, setupPath);
     }
 
-    if (remoteToken) {
-      args.push('-t', remoteToken);
+    if (metadata.productName) {
+      const msiPath = path.join(outputDirectory, `${metadata.productName}Setup.msi`);
+      const unfixedMsiPath = path.join(outputDirectory, 'Setup.msi');
+      if (await fsUtils.fileExists(unfixedMsiPath)) {
+        log(`Renaming ${unfixedMsiPath} => ${msiPath}`);
+        await fsUtils.rename(unfixedMsiPath, msiPath);
+      }
     }
-
-    log(await spawn(cmd, args));
   }
+}
+
+function pack(metadata, appDirectory, outFile) {
```

Comment:
No way to use 7z here because 7z doesn't support directory prefix. Anyway, we don't nee to compress here, so, node implementation is ok.